### PR TITLE
Trigger Remote Node checks from Batch Check API for Firewall ports

### PR DESCRIPTION
**Pull request by daveaugustus**
### :nut_and_bolt: Description: What code changed, and why?
Trigger Remote Node checks from Batch Check API for Firewall ports

### :chains: Related Resources

https://chefio.atlassian.net/browse/CHEF-2868

### :+1: Definition of Done
Remote Node execution:
- Checks firewall settings

### :athletic_shoe: How to Build and Test the Change
Run `make verify-serve` from `components/automate-cli/pkg/verifyserver` dir
And trigger the API

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Screenshot 2023-05-19 at 1 57 42 AM

### CODE REVIEW

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+// The request response is being constructed based on the https://docs.chef.io/automate/ha_on_premises_deployment_prerequisites/#firewall-checks (Firewall Checks)
+func makeRequests(config models.Config) []trigger.ReqBody {
+	var reqBodies []trigger.ReqBody
+
+	// _____________________________________________ Row 1: Chef Automate to all the OS and PG nodes _____________________________________________
+	// Dest postgres
+	for _, sourceNodeIP := range config.Hardware.AutomateNodeIps {
+		for _, destNodeIP := range config.Hardware.PostgresqlNodeIps {
+			reqBody := trigger.ReqBody{
+				NodeType:                   constants.AUTOMATE,
```
Comment (daveaugustus) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:45]: @Dmaddu will source will be the node type or destination?

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+				SourceNodeIP:               sourceNodeIP,
+				DestinationNodeIP:          destNodeIP,
+				DestinationServicePort:     "7432",
+				DestinationServiceProtocol: "tcp",
+				Cert:                       config.Certificate.Nodes[0].Cert,
```
Comment (daveaugustus) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:50]: How can we assign certificate for now I'm taking the 0th index but we can modify it?

cc: @Dmaddu

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+				Key:                        config.Certificate.Nodes[0].Key,
```
Comment (daveaugustus) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:51]: How can we assign Key for now I'm taking the 0th index but we can modify it?

cc: @Dmaddu

**Comment by YashviJain01** [comment]
Screenshot 2023-05-23 at 10 43 10 AM

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+			reqBody := models.FirewallRequest{
+				SourceNodeIP:               constants.LOCAL_HOST,
+				DestinationNodeIP:          destNodeIP,
+				DestinationServicePort:     port,
+				DestinationServiceProtocol: "tcp",
+			}
+			reqBodies = append(reqBodies, reqBody)
+		}
+	}
+	// Dest Chef Infra
```
Comment (kalroy) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:241]:
```suggestion
	// Dest OpenSearch
```

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+		}
+	}
+
+	return reqBodies
+}
+
+func getRequestAsBastionSource(config models.Config) []models.FirewallRequest {
+	var reqBodies []models.FirewallRequest
+
+	for _, destNodeIP := range config.Hardware.AutomateNodeIps {
```
Comment (kalroy) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:203]:
```suggestion
	// Dest Automate 
	for _, destNodeIP := range config.Hardware.AutomateNodeIps {
```

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+	}
+
+	for i := 0; i < reqCount; i++ {
+		select {
+		case res := <-outputCh:
+			result = append(result, res)
+		}
+	}
+
+	return result
```
Comment (kalroy) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:356]: Close the channel after the loop

**Review by kalroy — CHANGES_REQUESTED** [review]

**Review by kalroy — CHANGES_REQUESTED** [review]
Add tests for failure scenarios too.

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go [hunk 7]
```diff
@@ -0,0 +1,357 @@
...
+
+	return reqBodies
+}
+
+// getRequestsForPostgresAsSource gives the requests for all the ports where postgres is source
+func getRequestsForPostgresAsSource(config models.Config) []models.FirewallRequest {
+	var reqBodies []models.FirewallRequest
+	for _, sourceNodeIP := range config.Hardware.PostgresqlNodeIps {
+		for _, destNodeIP := range config.Hardware.PostgresqlNodeIps {
+			if sourceNodeIP != destNodeIP {
```
Comment (Dmaddu) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:144]: This logic is correct but for better Cyclomatic complexity we can write as below

if sourceNodeIP == destNodeIP {
continue
}
//rest of the logic.

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go [hunk 8]
```diff
@@ -0,0 +1,357 @@
...
+
+	return reqBodies
+}
+
+// getRequestsForOpensearchAsSource gives the requests for all the ports where opensearch is source
+func getRequestsForOpensearchAsSource(config models.Config) []models.FirewallRequest {
+	var reqBodies []models.FirewallRequest
+	for _, sourceNodeIP := range config.Hardware.OpenSearchNodeIps {
+		for _, destNodeIP := range config.Hardware.OpenSearchNodeIps {
+			if sourceNodeIP != destNodeIP {
```
Comment (Dmaddu) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:174]: same thing here also

**Review by Dmaddu — CHANGES_REQUESTED** [review]

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go (hunk 5 above)
Comment (YashviJain01) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:203]: Added relevant comments

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go (hunk 7 above)
Comment (YashviJain01) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:144]: Changed the logic

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go (hunk 8 above)
Comment (YashviJain01) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:174]: changes the logic

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go (hunk 6 above)
Comment (YashviJain01) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go:356]: missed on my side, corrected it now,

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go (hunk 8 above)
Comment (YashviJain01) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/firewallchecktrigger/firewallchecktrigger.go]: Added the logic

**Review by kalroy — APPROVED** [review]

**Review by Dmaddu — APPROVED** [review]

**Pull request merged**